shell/core/types.ts

```diff
@@ -567,3 +567,8 @@ export interface IPlugin {
    */
   DSL(store: any, productName: string): DSLReturnType;
 }
+
+// Internal interface
+// Built-in extensions may use this, but external extensions should not, as this is subject to change
+// Defined as any for now
+export type IInternal = any;
```

Comment:
We need to work this out properly - I've left this for now, as this will always be built-in, but need to come back to this when we improve the Extensions API/SDK.